### ISSUE TITLE
Add or improve documentation

### ISSUE THREAD

**Issue by loreloc** [april-tools/cirkit#367]
- [ ] Document graph and diacyclic graph data structures
- [ ]

**Comment by andreasgrv** [comment]
* [] Explain how to ignore certain outputs in cirkit (i.e. marginalise). E.g. how to handle the case where in pytorch with cross-entropy loss you would use -100.
